### PR TITLE
Feature/mx 1635 model update biospecimen

### PR DESCRIPTION
**Pull request by erichesse**
# PR Context
- model update biospecimen
- created [MX-1695] for less confusing wikidata querying

### CODE REVIEW

File: mex/extractors/biospecimen/main.py [hunk 1]
```diff
@@ -66,8 +69,19 @@ def extracted_biospecimen_resources(
     unit_stable_target_ids_by_synonym: dict[str, MergedOrganizationalUnitIdentifier],
     extracted_organization_rki: ExtractedOrganization,
     extracted_synopse_activities: list[ExtractedActivity],
+    extracted_primary_source_wikidata: ExtractedPrimarySource,
 ) -> list[ExtractedResource]:
     """Transform biospecimen resources to extracted resources and load them to the sinks."""  # noqa: E501
+    settings = Settings.get()
+    resource_mapping = extract_mapping_data(
+        settings.biospecimen.mapping_path / "resource.yaml", ExtractedResource
```
Comment (esinsj) [inline mex/extractors/biospecimen/main.py:77]: Do I understand correctly, that this new piece of code automatically reads in and implements mapping rules with default mapping values from the yaml-file for the mappings? Cool :)

File: mex/extractors/biospecimen/transform.py [hunk 2]
```diff
@@ -68,10 +76,22 @@ def transform_biospecimen_resource_to_mex_resource(
...
+                extracted_organizations,
+                extracted_primary_source_biospecimen,
+            )
+            if resource.externe_partner
+            else []
         )
+        has_personal_data = resource_mapping["hasPersonalData"][0]["mappingRules"][0][
+            "setValues"
+        ]
+        has_legal_basis = resource_mapping["hasLegalBasis"][0]["mappingRules"][0][
```
Comment (esinsj) [inline mex/extractors/biospecimen/transform.py:90]: Does this work? The setValues for "hasLegalBasis" has further subcategories: language and value (with language being a different field than the set language in the next ow 93.)

File: mex/extractors/biospecimen/transform.py [hunk 3]
```diff
@@ -89,42 +109,64 @@ def transform_biospecimen_resource_to_mex_resource(
             )
         else:
             documentation = None
+        loinc_id = (
```
Comment (esinsj) [inline mex/extractors/biospecimen/transform.py:112]: This should've already been implemented before. Did you check all mapping rules for completeness?

File: mex/extractors/biospecimen/transform.py [hunk 4]
```diff
@@ -89,42 +109,64 @@ def transform_biospecimen_resource_to_mex_resource(
...
         unit_in_charge = unit_stable_target_ids_by_synonym.get(
             resource.verantwortliche_fachabteilung
         )
-        theme = [
-            Theme["LABORATORY"],
-            Theme["INFECTIOUS_DISEASES"],
-            Theme["PHYSICAL_HEALTH"],
-        ]
+        if (
+            resource_mapping["theme"][0]["mappingRules"][1]["forValues"]
```
Comment (esinsj) [inline mex/extractors/biospecimen/transform.py:138]: Couldn't the indices shift? Wouldn't it then be safer to refer to them by addressing to the correct value by name?

File: mex/extractors/biospecimen/transform.py [hunk 5]
```diff
@@ -89,42 +109,64 @@ def transform_biospecimen_resource_to_mex_resource(
...
+            in resource.thema
+        ):
+            theme = resource_mapping["theme"][0]["mappingRules"][1]["setValues"]
+        else:
+            theme = resource_mapping["theme"][0]["mappingRules"][0]["setValues"]
         yield ExtractedResource(
-            accessRestriction=AccessRestriction["RESTRICTED"],
```
Comment (esinsj) [inline mex/extractors/biospecimen/transform.py:109]: Wouldn't it always be "Open" with this implementation, even if the value in the source is "restriktiv"?

File: mex/extractors/biospecimen/transform.py (hunk 4 above)
Comment (erichesse) [reply mex/extractors/biospecimen/transform.py:138]: What do you mean "by name"? There are two `mappingRules` in `theme`, one with index 0 and the other with index 1. So this is correct reference. 
We assume the mappings to be fairly stable and now established a workflow, that every time a mapping is changed, a developer ensures, that the code is in sync. But suggestions for making more reliable mappings are always welcomed.

File: mex/extractors/biospecimen/transform.py (hunk 5 above)
Comment (erichesse) [reply mex/extractors/biospecimen/transform.py:109]: I didn't even look at it as this field wasn't changed in the model update, however I agree, that the mapping rule is a bit confusing here. I made a comment on this issue in the wrap up ticket MX-1689

File: mex/extractors/biospecimen/transform.py [hunk 6]
```diff
@@ -136,3 +178,56 @@ def transform_biospecimen_resource_to_mex_resource(
             unitInCharge=unit_in_charge,
             wasGeneratedBy=was_generated_by or None,
         )
+
+
+def get_merged_organization_id_by_query_with_transform_and_load(
```
Comment (esinsj) [inline mex/extractors/biospecimen/transform.py:183]: this function already exists in mex.extractors.wikidata.extract. 
from mex.extractors.wikidata.extract import (
    get_merged_organization_id_by_query_with_transform_and_load,
)

Comment (erichesse) [reply mex/extractors/biospecimen/transform.py]: good point, fixed that: https://github.com/robert-koch-institut/mex-extractors/pull/207/commits/45fbff37f3b8963259079d5f9af3eb791be4d283

File: mex/extractors/biospecimen/transform.py (hunk 4 above)
Comment (esinsj) [reply mex/extractors/biospecimen/transform.py:133]: you're right.

**Review by esinsj — APPROVED** [review]
Thanks for the replies. I didn't find anything else.

**Review by esinsj — APPROVED** [review]

**Comment by esinsj** [comment]
ready to merge imho

**Pull request merged**